File: services/auth-service/src/main/java/com/otterworks/auth/dto/UserSettingsDTO.java
package com.otterworks.auth.dto;

import com.otterworks.auth.entity.UserSettings;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserSettingsDTO {
  private Boolean notificationEmail;
  private Boolean notificationInApp;
  private Boolean notificationDesktop;
  private String theme;
  private String language;

  public static UserSettingsDTO fromEntity(UserSettings entity) {
    return new UserSettingsDTO(
        entity.isNotificationEmail(),
        entity.isNotificationInApp(),
        entity.isNotificationDesktop(),
        entity.getTheme(),
        entity.getLanguage());
  }

  public static UserSettingsDTO defaults() {
    return new UserSettingsDTO(true, true, false, "system", "en");
  }
}

File: services/auth-service/src/DTOs/UserSettingsDTO.cs
using OtterWorks.AuthService.Data.Entities;

namespace OtterWorks.AuthService.DTOs;

public sealed class UserSettingsDTO
{
    public bool NotificationEmail { get; set; }
    public bool NotificationInApp { get; set; }
    public bool NotificationDesktop { get; set; }
    public string Theme { get; set; } = string.Empty;
    public string Language { get; set; } = string.Empty;

    public static UserSettingsDTO FromEntity(UserSettings entity)
    {
        return new UserSettingsDTO
        {
            NotificationEmail = entity.NotificationEmail,
            NotificationInApp = entity.NotificationInApp,
            NotificationDesktop = entity.NotificationDesktop,
            Theme = entity.Theme,
            Language = entity.Language,
        };
    }
}
